src/safeds/data/labeled/containers/_time_series_dataset.py

```diff
@@ -0,0 +1,323 @@
+        """The feature columns of the time series dataset."""
+        return self._features
+
+    @property
+    def target(self) -> Column:
+        """The target column of the time series dataset."""
+        return self._target
+
+    @property
+    def time(self) -> Column:
+        """The time column of the time series dataset."""
+        return self._time
+
+    @property
+    def extras(self) -> Table:
+        """
+        Additional columns of the time series dataset that are neither features, target nor time.
+
+        These can be used to store additional information about instances, such as IDs.
+        """
+        return self._extras
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Conversion
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def to_table(self) -> Table:
+        """
+        Return a new `Table` containing the feature columns and the target column.
+
+        The original `TabularDataset` is not modified.
+
+        Returns
+        -------
+        table:
+            A table containing the feature columns and the target column.
+        """
+        return self._table
+
+    def _into_dataloader_with_window(self, window_size: int, forecast_horizon: int, batch_size: int) -> DataLoader:
+        """
+        Return a Dataloader for the data stored in this time series, used for training neural networks.
+
+        It splits the target column into windows, uses them as feature and creates targets for the time series, by
+        forecast length. The original table is not modified.
+
+        Parameters
+        ----------
+        window_size:
+            The size of the created windows
+
+        forecast_horizon:
+            The length of the forecast horizon, where all datapoints are collected until the given lag.
+
+        batch_size:
+            The size of data batches that should be loaded at one time.
+
+
+        Returns
+        -------
+        result:
+            The DataLoader.
+        """
+        import numpy as np
+        from torch.utils.data import DataLoader
+
+        target_np = self.target._data.to_numpy()
+
+        x_s = []
+        y_s = []
+
+        size = len(target_np)
+        # create feature windows and for that features targets lagged by forecast len
+        # every feature column wird auch gewindowed
+        # -> [i, win_size],[target]
+        feature_cols = self.features.to_columns()
+        for i in range(size - (forecast_horizon + window_size)):
+            window = target_np[i : i + window_size]
+            label = target_np[i + window_size + forecast_horizon]
+            for col in feature_cols:
+                data = col._data.to_numpy()
+                window = np.concatenate((window, data[i : i + window_size]))
+            x_s.append(window)
+            y_s.append(label)
```

Comment:
Could you do this conversion directly in pytorch without the use of numpy? This could lead to faster conversion, especially when the user uses a gpu. The same question applies to all other conversions.

Reply:
yes, added